tools/payments/client.go

```diff
@@ -0,0 +1,210 @@
+		env: env,
+		redis: redis.NewClusterClient(
+			&redis.ClusterOptions{
+				Addrs: strings.Split(addrs, ","), Password: pass, Username: username,
+				DialTimeout:     15 * time.Second,
+				WriteTimeout:    5 * time.Second,
+				MaxRetries:      5,
+				MinRetryBackoff: 5 * time.Millisecond,
+				MaxRetryBackoff: 500 * time.Millisecond,
+				PoolSize:        10,
+				PoolTimeout:     30 * time.Second,
+				TLSConfig:       tlsConfig,
+			}),
+	}
+	err := rc.redis.Ping(ctx).Err()
+	if err != nil {
+		return nil, fmt.Errorf("failed to ping redis: %w", err)
+	}
+	return rc, nil
+}
+
+// ConfigureWorker implements settlement client
+func (rc *redisClient) ConfigureWorker(ctx context.Context, stream string, config *WorkerConfig) error {
+	body, err := json.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to json encode config: %w", err)
+	}
+
+	cfg := &prepareWrapper{
+		ID:        uuid.New(),
+		Timestamp: time.Now(),
+		Body:      string(body),
+	}
+
+	_, err := rc.redis.XAdd(
+		ctx, &redis.XAddArgs{
+			Stream: stream,
+			Values: map[string]interface{}{
+				"data": cfg}},
+	).Result()
+	if err != nil {
+		return fmt.Errorf("failed to push config to workers: %w", err)
+	}
+	return nil
+}
+
+// PrepareTransactions implements settlement client
+func (rc *redisClient) PrepareTransactions(ctx context.Context, t ...*PrepareTx) error {
+	pipe := rc.redis.Pipeline()
+
+	for _, v := range t {
+		body, err := json.Marshal(v)
+		if err != nil {
+			return fmt.Errorf("failed to serialize transaction: %w", err)
+		}
+
+		// message wrapper for prepare
+		message := &prepareWrapper{
+			ID:        uuid.New(),
+			Timestamp: time.Now(),
+			Body:      string(body),
+		}
+
+		// add to stream
+		_, err = pipe.XAdd(
+			ctx, &redis.XAddArgs{
+				Stream: prepareStream,
+				Values: map[string]interface{}{
+					"data": message}},
+		).Result()
+		if err != nil {
+			return fmt.Errorf("failed to prepare transaction: %w", err)
+		}
+	}
+	return nil
+}
+
+// SubmitTransactions implements settlement client
+func (rc *redisClient) SubmitTransactions(ctx context.Context, signer httpsignature.ParameterizedSignator, at ...*AttestedTx) error {
+	pipe := rc.redis.Pipeline()
+
+	for _, v := range at {
+		buf := bytes.NewBuffer([]byte{})
+		err := json.NewEncoder(buf).Encode(v)
+		body := buf.Bytes()
+		if err != nil {
+			return fmt.Errorf("failed to marshal attested transaction body: %w", err)
+		}
+		// create a request
+		req, err := http.NewRequest("POST", rc.env+"/authorize", buf)
+		if err != nil {
+			return fmt.Errorf("failed to create request to sign: %w", err)
+		}
+		// we will be signing, need all these headers for it to go through
+		req.Header.Set(hostHeader, rc.env)
+		req.Header.Set(digestHeader, fmt.Sprintf("%x", sha256.Sum256(body)))
+		req.Header.Set(contentLengthHeader, fmt.Sprintf("%d", len(body)))
+		req.Header.Set(contentTypeHeader, "application/json")
+
+		// http sign the request
+		err = signer.SignRequest(req)
```

Comment:
This is more relevant to the services side, but calling it out now to make sure we're addressing it. We should be double checking the requests aren't being replayed in any sort of way. I believe that the authorization portion of this flow would prevent this because the transaction submissions should be uniquely identified.